src/models/data.py

```diff
@@ -229,17 +247,20 @@ def __next__(self) -> Tuple[Union[np.ndarray, torch.Tensor],
 
                     cur_max_idx = min(cur_max_idx + 1, self.max_idx)
 
-                out_x.append(arrays.x)
+                out_x.append(arrays.x.historical)
+                out_x_add.append(arrays.x.pred_months)
                 out_y.append(arrays.y)
                 self.idx += 1
 
             final_x = np.concatenate(out_x, axis=0)
+            final_x_add = np.concatenate(out_x_add, axis=0)
             final_y = np.concatenate(out_y, axis=0)
             if final_x.shape[0] == 0:
                 raise StopIteration()
             if self.to_tensor:
-                return torch.from_numpy(final_x).float(), torch.from_numpy(final_y).float()
-            return final_x, final_y
+                return (torch.from_numpy(final_x).float(),
+                        torch.from_numpy(final_x_add).float()), torch.from_numpy(final_y).float()
```

Comment:
what if there is no additional data (as in our vanilla models?)

Reply:
The iterator will always return the pred months, but the models can decide not to use them (if `include_pred_month` is False)